lib/active_delivery/base.rb

```diff
@@ -73,6 +73,10 @@ def #{line_id}_class
         CODE
       end
 
+      def unregister_line(line_id)
+        delivery_lines.delete(line_id)
```

Comment:
I think, we also need to remove the corresponding auto-generated methods (see above).

Reply:
Ah — okay will do that.

Reply:
Okay spent a moment looking and thinking about this.  Using the rspec example code as a talking point: we have a method added to `ActiveDelivery::Base.quack_quack_class` when we register the `:quack_quack` line.  

Next we have a `MyDelivery`, which inherits from `ActiveDelivery::Base`.  And in `MyDelivery` we want to remove the `:quack_quack` delivery line so that it doesn't run when `MyDelivery` is sending to each line.  

If we go ahead and actually _remove_ the corresponding `.quack_quack_class` method when `MyDeliver.unregister_line` is called we would have to remove it **from** `ActiveDelivery::Base` (where it is defined) ... which I don't think we want to do because other deliveries would likely inherit from it from this class as well and expect to find it there.  We just want `MyDelivery` to unregister that line.

So if the `MyDelivery` does "own" those class methods, then should we allow those methods to still appear to exist (even though the line will not be run)?  Or should we raise `NoMethodError` on `MyDelivery` somehow manually overriding the expected behavior?

I think if `unregister_line` is called on `ActiveRecord::Base` (again in this example from the test) then we _do_ want to remove the methods because that's where they were defined.

Hope I am making sense!

Reply:
Oh, you want to remove a line from a specific delivery, not globally.

Then we can use `undef_method`—that's what it is for :)